File: src/main/java/com/github/nsc/de/compiler/parser/node/VariableUsageNode.java
package com.github.nsc.de.compiler.parser.node;

public class VariableUsageNode implements Node {
    private final String name;

    public VariableUsageNode(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }

    @Override
    public String toString() {
        return name;
    }
}

File: parser/src/commonMain/kotlin/com/github/shakelang/shake/parser/node/variables/VariableUsageNode.kt
package com.github.shakelang.shake.parser.node.variables

import com.github.shakelang.shake.parser.node.ValuedNode
import com.github.shakelang.shake.parser.node.IdentifierNode
import com.github.shakelang.shake.util.characterinput.position.PositionMap

class VariableUsageNode(map: PositionMap, val variable: IdentifierNode) : ValuedNode(map) {
    override fun toJson(): Map<String, *> = mapOf("name" to "VariableUsageNode", "variable" to variable.json)
}